### PR TITLE
Support LoadBalancer child context creation during AOT processing

### PR DESCRIPTION
**Pull request by OlgaMaciaszek**

### CODE REVIEW

File: spring-cloud-context/src/main/java/org/springframework/cloud/context/named/NamedContextFactory.java [hunk 1]
```diff
@@ -126,33 +161,20 @@ protected AnnotationConfigApplicationContext createContext(String name) {
...
+			context.setClassLoader(parent.getClassLoader());
 		}
 		else {
-			context = new AnnotationConfigApplicationContext();
-		}
-		if (this.configurations.containsKey(name)) {
-			for (Class<?> configuration : this.configurations.get(name).getConfiguration()) {
-				context.register(configuration);
-			}
+			context = new GenericApplicationContext();
```
Comment (spencergibb) [inline spring-cloud-context/src/main/java/org/springframework/cloud/context/named/NamedContextFactory.java:169]: Should this get the AotDetector treatment as well?

File: spring-cloud-loadbalancer/pom.xml [hunk 2]
```diff
@@ -128,6 +128,11 @@
 			<artifactId>reactor-test</artifactId>
 			<scope>test</scope>
 		</dependency>
+		<dependency>
+			<groupId>org.springframework</groupId>
+			<artifactId>spring-core-test</artifactId>
```
Comment (spencergibb) [inline spring-cloud-loadbalancer/pom.xml:133]: Interesting. Doesn't spring-boot-starter-test bring this in?

**Review by spencergibb — APPROVED** [review]
LGTM, a couple of small questions.

File: spring-cloud-context/src/main/java/org/springframework/cloud/context/named/NamedContextFactory.java (hunk 1 above)
Comment (OlgaMaciaszek) [reply spring-cloud-context/src/main/java/org/springframework/cloud/context/named/NamedContextFactory.java:169]: Good catch. I think for child contexts we always have the parent, but maybe there's some edge-case where we don't... I guess better to have it there as well.

**Pull request merged**